### PR TITLE
Improve test suite and CI run performance

### PR DESCRIPTION
**Pull request by benjaminwil**
This pull request makes our GitHub Actions test suite workflow run faster. I took this opportunity to also simplify the workflow YAML file.

I started doing this work because of JRuby CI failures. (See #1779.) But then noticed some low-hanging fruit that would improve our test suite runs and maintenance in general.

### CODE REVIEW

**Comment by benjaminwil** [comment]
If there is not a good reason to continue support Ruby 2.4, we can also drop 2.4 from CI runs as part of this PR. See #1795.

File: test/test_app.rb [hunk 1]
```diff
@@ -39,31 +39,29 @@
...
-    page = 'utfh1'
-    text = nfd('한글')
+    page_content = <<~TEXT
+      ## 한글
 
-    # don't use h1 or it will be promoted to replace file name
-    # which doesn't generate a normal header link
-    @wiki.write_page(page, :markdown, '## ' + text,
-                     { :name => 'user1', :email => 'user1' });
+      Test page "utfh1" content.
```
Comment (dometto) [inline test/test_app.rb:51]: Much more readable, thanks!

File: test/test_app.rb [hunk 2]
```diff
@@ -39,31 +39,29 @@
     assert_match /<pre><code>one\ntwo\nthree\nfour\n<\/code><\/pre>\n/m, last_response.body
   end
 
-  def nfd utf8
```
Comment (dometto) [inline test/test_app.rb:42]: I think this means we can remove `twitter_cldr` as a development dependency from `gollum.gemspec`, too.

**Comment by dometto** [comment]
See comments, but looks great! Fine to also remove 2.4 testing for further simplification

File: test/test_app.rb (hunk 2 above)
Comment (benjaminwil) [reply test/test_app.rb:42]: Oh wow! Thanks for noticing that.

**Pull request merged**